[BUG]: CCCL 3.1 fails to compile with libc++: no type named 'value_type' in 'std::iterator_traits<int[]>'

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

libcu++

### Describe the bug

libcudacxx/test/libcudacxx/std/iterators/iterator.requirements/iterator.assoc.types/readable.traits/iter_value_t.pass.cpp fails to compile with libc++ (failure happens with both nvcc and clang):

https://godbolt.org/z/8b4ace3Ws

```
In file included from <source>:2:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/iterator:25:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/advance.h:26:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/concepts.h:41:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/iter_move.h:25:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/iterator_traits.h:35:
/opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/readable_traits.h:149:1: error: no type named 'value_type' in 'std::iterator_traits<int[]>'
  149 | using iter_value_t =
      | ^~~~~
<source>:7:57: note: in instantiation of template type alias 'iter_value_t' requested here
    7 |   constexpr bool result = cuda::std::same_as<cuda::std::iter_value_t<T>, Expected>;
      |                                                         ^
<source>:22:15: note: in instantiation of function template specialization 'check_iter_value_t<int[], int>' requested here
   22 | static_assert(check_iter_value_t<int[], int>(), "");
      |               ^
In file included from <source>:2:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/iterator:25:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/advance.h:26:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/concepts.h:41:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/iter_move.h:25:
In file included from /opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/iterator_traits.h:35:
/opt/compiler-explorer/libs/cccl/trunk/libcudacxx/include/cuda/std/__iterator/readable_traits.h:149:1: error: no type named 'value_type' in 'std::iterator_traits<int[]>'
  149 | using iter_value_t =
      | ^~~~~
<source>:8:47: note: in instantiation of template type alias 'iter_value_t' requested here
    8 |   static_assert(cuda::std::same_as<cuda::std::iter_value_t<T const>, Expected> == result, "");
      |                                               ^
<source>:22:15: note: in instantiation of function template specialization 'check_iter_value_t<int[], int>' requested here
   22 | static_assert(check_iter_value_t<int[], int>(), "");
      |               ^
<source>:22:15: error: static assertion expression is not an integral constant expression
   22 | static_assert(check_iter_value_t<int[], int>(), "");
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```


### How to Reproduce

See godbolt link for the reproducer.



### Expected behavior

Compilation should work with both libstdc++ and libc++.

### Reproduction link

https://godbolt.org/z/8b4ace3Ws

### Operating System

_No response_

### nvidia-smi output

n/a

### NVCC version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Wed_Aug_20_01:58:59_PM_PDT_2025
Cuda compilation tools, release 13.0, V13.0.88
Build cuda_13.0.r13.0/compiler.36424714_0
nvcc warning : incompatible redefinition for option 'compiler-bindir', the last value of this option was used
Compiler returned: 0

--- a/libcudacxx/include/cuda/std/__type_traits/is_primary_template.h
+++ b/libcudacxx/include/cuda/std/__type_traits/is_primary_template.h
@@ -84,11 +84,13 @@ template <class _Iter>
 struct __is_primary_std_template : bool_constant<__is_primary_std_template_impl<_Iter, is_pointer_v<_Iter>>>
 {};
 #  elif defined(_LIBCPP_VERSION)
+
 // libc++ uses the same mechanism than we do with __primary_template
 template <class _Traits>
 using __test_for_primary_std_template = enable_if_t<_IsSame<_Traits, typename _Traits::__primary_template>::value>;
 template <class _Iter>
-using __is_primary_std_template = _IsValidExpansion<__test_for_primary_template, ::std::iterator_traits<_Iter>>;
+using __is_primary_std_template = _IsValidExpansion<__test_for_primary_std_template, ::std::iterator_traits<_Iter>>;
+
 #  elif defined(_MSVC_STL_VERSION) || defined(_IS_WRS)
 // On MSVC we must check for the base class because `_From_primary` is only defined in C++20
 template <class _Iter, bool>

